src/components/common/AddressInput/index.tsx

```diff
@@ -55,73 +76,86 @@ const AddressInput = ({ name, validate, required = true, deps, ...props }: Addre
     }
   }, [address, currentShortName, setAddressValue])
 
-  return (
-    <Grid container alignItems="center" gap={1}>
-      <Grid item flexGrow={1}>
-        <TextField
-          {...props}
-          autoComplete="off"
-          label={<>{error?.message || label}</>}
-          error={!!error}
-          fullWidth
-          spellCheck={false}
-          InputProps={{
-            ...(props.InputProps || {}),
-
-            // Display the current short name in the adornment, unless the value contains the same prefix
-            startAdornment: !error && !rawValueRef.current.startsWith(`${currentShortName}:`) && (
-              <InputAdornment position="end">{currentShortName}:</InputAdornment>
-            ),
-
-            endAdornment: (resolving || isValidating) && (
-              <InputAdornment position="end">
-                <CircularProgress size={20} />
-              </InputAdornment>
-            ),
-          }}
-          InputLabelProps={{
-            ...(props.InputLabelProps || {}),
-            shrink: !!watchedValue || props.focused,
-          }}
-          required={required}
-          {...register(name, {
-            deps,
-
-            required,
-
-            setValueAs: (value: string): string => {
-              // Clean the input value
-              const cleanValue = cleanInputValue(value)
-              rawValueRef.current = cleanValue
-              // This also checksums the address
-              return parsePrefixedAddress(cleanValue).address
-            },
-
-            validate: async () => {
-              const value = rawValueRef.current
-              if (value) {
-                setIsValidating(true)
-                const result = validatePrefixed(value) || (await validate?.(parsePrefixedAddress(value).address))
-                setIsValidating(false)
-                return result
-              }
-            },
-
-            // Workaround for a bug in react-hook-form that it restores a cached error state on blur
-            onBlur: () => setTimeout(() => trigger(name), 100),
-          })}
-          // Workaround for a bug in react-hook-form when `register().value` is cached after `setValueAs`
-          // Only seems to occur on the `/load` route
-          value={watchedValue}
-        />
-      </Grid>
-
-      {!props.disabled && (
-        <Grid item>
+  const endAdornment = (
+    <InputAdornment position="end">
+      {resolving || isValidating ? (
+        <CircularProgress size={20} />
+      ) : !props.disabled ? (
+        <>
+          <IconButton onClick={() => setOpen(true)}>
+            <SvgIcon component={SaveAddressIcon} inheritViewBox fontSize="small" color="primary" />
+          </IconButton>
```

Comment:
Nit: I would not show this if the address is already present in the address book.

Reply:
If the address is present in the AB, it renders the `AddressInputReadOnly` and we only show the button there if the address is not contained in the AB.

Reply:
I see. It probably makes more sense if we use the read only flag and don't keep the `AddressInputReadOnly` component then.